mypy/subtypes.py

```diff
@@ -834,43 +833,36 @@ def visit_overloaded(self, left: Overloaded) -> bool:
                     # Order matters: we need to make sure that the index of
                     # this item is at least the index of the previous one.
                     if subtype_match and previous_match_left_index <= left_index:
-                        if not found_match:
-                            # Update the index of the previous match.
-                            previous_match_left_index = left_index
-                            found_match = True
-                            matched_overloads.add(left_item)
-                            possible_invalid_overloads.discard(left_item)
+                        previous_match_left_index = left_index
+                        found_match = True
+                        matched_overloads.add(left_index)
+                        break
```

Comment:
adding the break here is the main functional change, everything in the second commit is non-functional